tests/commands_test.rs

```diff
@@ -146,12 +165,16 @@ fn cp_error_if_attempting_to_copy_a_directory_to_another_directory() {
 
 #[test]
 fn rm_can_remove_a_file() {
-    let directory = "tests/fixtures/nuplayground";
+    let directory   = "tests/fixtures/nuplayground";
     let file = format!("{}/rm_test.txt", directory);
 
     h::create_file_at(&file);
 
-    nu!(_output, cwd(directory), "rm rm_test.txt");
+    nu!(
+        _output,
+        cwd(directory),
+        "rm rm_test.txt"
```

Comment:
You know, I was wondering looking at this:

When we're doing operations like copying and removing (or saving), what directories are we writing into? We might want to do that kind of file manipulation in a temporary directory *just in case* something goes wrong on the users system, we have less chance of deleting something in the source directories.

Reply:
In the helper module there is `setup_playground_for` that accepts a "topic" (the name of the directory where the test case will create inside the `playground` directory) 

For instance, you changed the other test case to `rm_test2` due to race condition.

so if you do `h::setup_playground_for("rm_test_2")` it will create the directory (deletes it first if it exists already that happens when the test case ends and this is for inspection purposes in case the test fails)

so any test case that has to deal with copying, removing, saving will have a unique directory inside `nuplayground`

`h::setup_playground_for("rm_test_2") => tests/fixtures/nuplayground/rm_test_2`
`h::setup_playground_for("save_test_1") => tests/fixtures/nuplayground/save_test_1`

Reply:
So everytime we do a `cargo test` all those playgrounds get deleted and recreated so we can have a blank slate.

In the case of this save test it creates a directory under `/tests/fixtures/nuplayground/save_smart_test`

then copies `tests/fixtures/formats/cargo_sample.toml` into `/tests/fixtures/nuplayground/save_smart_test/cargo_sample.toml`

then nu goes `open nuplayground/save_smart_test/cargo_sample.toml | inc package.version --minor | save`

and we check the saved file there.

Reply:
We never touch the files `under tests/fixtures/formats`. All the mutations happen directly in `tests/fixtures/nuplayground`where each test case will have a unique directory (`save_test`, `save_test2`, `rm_test`, etc)

Reply:
yes, we do everything in temporary directories under `nuplayground`.

Reply:
Got it! Thanks for the explanation. Yeah, that sounds good.